client/src/components/GraphViewerComponent/GraphViewerRelationshipCreateComponent/GraphViewerRelationshipCreateComponent.js

```diff
@@ -93,56 +106,98 @@ export class GraphViewerRelationshipCreateComponent extends Component {
+      const relationshipItems = await new ModelService().getRelationships(source.modelId, target.modelId);
+      if (relationshipItems.length === 0) {
+        this.setState({hasRelationships: false});
+      } else {
+        this.setState({hasRelationships: true});
+      }
+      this.setState({ relationshipItems });
+    } catch (exc) {
+      this.setState({ relationshipItems: [] });
+      exc.customMessage = `Error in retrieving model. Requested ${this.state.sourceModelId}`;
+      eventService.publishError(exc);
+    }
+    this.setState({ isLoading: false });
   }
 
   render() {
-    const { relationshipItems, relationshipId, isLoading, showModal, hasRequiredRelError } = this.state;
-    const { sourceId, targetId } = this.getNodes();
+    const { relationshipItems, relationshipId, isLoading, showModal, hasRequiredRelError, sourceId, targetId } = this.state;
 
     return (
       <ModalComponent isVisible={showModal} isLoading={isLoading} className="gc-dialog">
         <h2 className="heading-2" tabIndex="0" id="create-relationship-heading">Create Relationship</h2>
         <h4>Source ID</h4>
         <TextField disabled readOnly id="sourceIdField" ariaLabel="Source ID" className="modal-input" styles={this.getStyles} value={sourceId} />
+        <div className="btn-icon" style={{ float: "right" }}>
+          <IconButton iconOnly="true" iconProps={this.swapRelationshipIcon} title="Swap Relationship" ariaLabel="Swap Relationship" onClick={this.swap} />
+        </div>
         <h4>Target ID</h4>
         <TextField disabled readOnly id="targetIdField" ariaLabel="Target ID" className="modal-input" styles={this.getStyles} value={targetId} />
         <h4>Relationship</h4>
-        <Dropdown
-          tabIndex="0"
-          ariaLabel="Select an option"
-          required
-          placeholder="Select an option"
-          className="modal-input"
-          selectedKey={relationshipId}
-          options={relationshipItems.map((q, i) => ({ key: i, text: q }))}
-          styles={{
-            dropdown: { width: 208 }
-          }}
-          errorMessage={hasRequiredRelError ? "Please select a relationship" : null}
-          onChange={this.onSelectedRelChange} />
+        <div style={{ display: this.state.hasRelationships ? "block" : "none" }}>
+          <Dropdown
+            tabIndex="0"
+            ariaLabel="Select an option"
+            required
+            placeholder="Select an option"
+            className="modal-input"
+            selectedKey={relationshipId}
+            options={relationshipItems.map((q, i) => ({ key: i, text: q }))}
+            styles={{
+              dropdown: { width: 208 }
+            }}
+            errorMessage={hasRequiredRelError ? "Please select a relationship" : null}
+            onChange={this.onSelectedRelChange} />
+        </div>
+        <div style={{ backgroundColor: "red", height: "60px", display: this.state.hasRelationships ? "none" : "block" }}>
```

Comment:
I'd prefer to use classes rather than inline styles. See above re: display alternative.

Reply:
Change complete for all occurrences